PMP : fix isotropic remeshing surviving tiny constraints

janetournois opened the pull request:
## Summary of Changes

Before collapsing a short edge, the remesher performs the test : `collapse_does_not_invert_face(halfedge)`, that checks whether the collapse would make the mesh fold on itself.

Testing orientation of normals per patch (identified by their Id) is not a
good solution, because it can happen that we are trying to collapse an edge
that is sharp but which has both of its incident faces on the same surface
patch (wrt ids).

The new test is a lot simpler : for each non-degenerate face of the link of
the edge to be collapsed, simply check if its own normal has changed
orientation or not. The adjacent faces do not need to be taken into account

Edit on March 8 : the function `collapse_does_not_invert_face(halfedge)` has been renamed to `collapse_would_invert_face(halfedge)`, and the meaning reversed

## Release Management

* Affected package(s): PMP

janetournois commented:
This PR is a bug fix, so I should rebase my branch on the 4.12 branch and change the milestone, right @lrineau ?
(i have to check my first commit is correct with 4.12)

lrineau commented:
Be careful Maxime, Jane force-pushed a new patch: https://github.com/CGAL/cgal/pull/3735#event-2186594288. Last night's tests are already outdated.

maxGimeno commented:
errors :/ https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-142.shtml#Polygon_mesh_processing

maxGimeno commented:
Will it still fail building in master ?

janetournois commented:
no it should be fine now!

maxGimeno commented:
Okay, yhank you !

maxGimeno commented:
I'm sorry but there is a non trivial conflict with master :/

janetournois commented:
It's too tricky to have the same PR with no conflicts for 4.12, so let's change the base branch to 4.13.
I rebased on the 4.13-branch, fixed the conflicts and force-pushed.

maxGimeno commented:
https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-143.shtml All green :)

janetournois commented:
Great! Do not merge this PR straight away though, we may have a simpler way to make the same test

janetournois commented:
This PR is ready for testing again!

maxGimeno commented:
I remove this PR form integration for now, so that the beta-2 can be validated asap. I'll put it back after that, hopefully tomorrow after-noon.

janetournois commented:
ok!

maxGimeno commented:
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-151.shtml
